Katello/katello
# Fixes #31616 - Remove content_types requirement in katello.yaml and use of it

This PR removes the need for configuring `content_types` by registering types based only only on what Pulp 3 plugins are installed. The `use_pulp_2_for_content_type` config option has also been removed.

I'm keeping this as a draft for now while I fix all of the tests.

To test:
1) Remove the entire `content_types` section from the Katello configuration.  Try using content types like normal.
2) Shutdown Katello, and remove all Pulp 3 capabilities from your smart proxy with the following:
```ruby
SmartProxy.pulp_primary.smart_proxy_features.where(:feature_id => Feature.find_by(:name => SmartProxy::PULP3_FEATURE).id).update(:capabilities => [])
```
3) Ensure that you can still use all content types like normal without intervention after starting up Katello
